core/dbt/contracts/graph/nodes.py

```diff
@@ -567,6 +571,7 @@ class HookNode(CompiledNode):
 class ModelNode(CompiledNode):
     resource_type: NodeType = field(metadata={"restrict": [NodeType.Model]})
     access: AccessType = AccessType.Protected
+    state_relation: Optional[RelationalNode] = None
```

Comment:
Given the approach of storing stateful information on each node about its prod counterpart, we do need a place on the node object to do that. For now, I'm adding a new attribute (default `None`) to models, seeds, and snapshots — the three `refable` nodes that are eligible for deferral & cloning.